rust_src/src/frames.rs

```diff
@@ -118,11 +118,13 @@ pub fn set_frame_selected_window(
 pub fn framep(object: LispObject) -> LispObject {
     if let Some(frame) = object.as_frame() {
         match unsafe { fget_output_method(frame.as_ptr()) } {
-            OutputMethod::output_initial => LispObject::constant_t(),
-            OutputMethod::output_termcap => LispObject::constant_t(),
-            OutputMethod::output_x_window => LispObject::from(Qx),
-            OutputMethod::output_w32 => LispObject::from(Qw32),
-            OutputMethod::output_ns => LispObject::from(Qns),
+            0 => LispObject::constant_t(),
+            1 => LispObject::constant_t(),
+            2 => LispObject::from(Qx),
+            3 => LispObject::from(Qw32),
+            4 => LispObject::from(Qpc),
+            5 => LispObject::from(Qns),
+            _ => panic!("Invalid output method"),
         }
```

Comment:
I don't know how to avoid the _ pattern ?

Reply:
You don't - that's the consequence of having unreliable enums in C.

But you should introduce some named constants for the existing cases.

Reply:
What about using `unreachable!` or implemented our own similar macro?

Reply:
@birkenfeld I wasn't sure if you really mean that. I will try to fix it.